### PR TITLE
Jb tues ticket8

### PR DESCRIPTION
**Pull request by jdbbaugh**
## Ticket #1 
Note this is for ticket 1 despite branch name indicating it's for ticket8... while messin with 8 I found some stuff that needed to be sorted for 1

Given a user is viewing any page on the Bangazon site
When the user performs a gesture on the Product Categories hyperlink
Then the user will see a view containing a list of all product categories
And next to the category name, the number of products in that category will be displayed
And the first three products in the category will be displayed beneath each category, that are hyperlinks to the product detail

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Testing Instructions for Change Made

1.  `git fetch --all`
1.  `git checkout jb-tuesTicket8`
1.  run Bangazon
1.  in nav bar click "Products"  (don't use dropdown)
1.  you will see product types displayed along with amount of that particular product type available
1.  if particular product type doesn't have product in it, you will not see that product type listed
1.  product types will list no more then 3 of it's available products
1.  now click an available product under any of the Product Types (like "kite")
1.  you will now see a detailed description of the kite (Note: Add to order button does not work)
1.  in detailed description "back to list" hyperlink will return you to product type list you came from

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by THATBLADEBOY — APPROVED** [review]
Works as described 👍

**Review by colebryant — APPROVED** [review]
Looks good

**Review by dakotacookson — APPROVED** [review]
Lovley

**Review by askingalot — CHANGES_REQUESTED** [review]
clean up specific category views/methods.

File: Bangazon/Controllers/OrdersController.cs [hunk 1]
```diff
@@ -81,12 +117,19 @@ public async Task<IActionResult> Edit(int? id)
...
-            ViewData["PaymentTypeId"] = new SelectList(_context.PaymentType, "PaymentTypeId", "AccountNumber", order.PaymentTypeId);
-            ViewData["UserId"] = new SelectList(_context.ApplicationUsers, "Id", "Id", order.UserId);
+
+            var user = await GetCurrentUserAsync();
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            ViewData["PaymentTypeId"] = new SelectList(_context.PaymentType.Where(p => p.UserId == user.Id), "PaymentTypeId", "PaymentMethod");
```
Comment (askingalot) [inline Bangazon/Controllers/OrdersController.cs:131]: Use a view model

File: Bangazon/Views/Orders/Cart.cshtml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        </tr>
+        <tr>
+            <td>
+                <div>
+                    <input type="button" value="Complete Order" class="btn btn-primary"
+                           onclick="location.href = '@Url.Action("Edit", "Orders", new {id = orderItem.OrderId})'" />
+                </div>
+                <br/>
+                <div>
+                    <input type="button" value="Delete Order" class="btn btn-danger"
```
Comment (askingalot) [inline Bangazon/Views/Orders/Cart.cshtml:50]: this is clever, but use an HTML `form`

**Review by askingalot — CHANGES_REQUESTED** [review]

**Pull request closed without merging**